akka-cluster/src/test/scala/akka/cluster/ClusterDomainEventSpec.scala

```diff
@@ -79,6 +79,52 @@ class ClusterDomainEventSpec extends WordSpec with Matchers {
       diffSeen(ClusterSettings.DefaultDataCenter, g1, g2, selfDummyAddress) should ===(Seq.empty)
     }
 
+    "be produced for unreachability observations from the same dc" in {
+      val dc2MemberUp = TestMember(Address("akka.tcp", "sys", "otm", 2552), Up, Set.empty, "dc2")
+      val dc2MemberDown = TestMember(Address("akka.tcp", "sys", "otm", 2552), Down, Set.empty, "dc2")
+
+      val reachability1 = Reachability.empty
+      val g1 = Gossip(members = SortedSet(aUp, dc2MemberUp), overview = GossipOverview(reachability = reachability1))
+
+      val reachability2 = reachability1
+        .unreachable(aUp.uniqueAddress, dc2MemberDown.uniqueAddress)
+      val g2 = Gossip(members = SortedSet(aUp, dc2MemberDown), overview = GossipOverview(reachability = reachability2))
+
+      diffUnreachableTeam(g1, g2, aUp.uniqueAddress) should ===(Seq(UnreachableDataCenter("dc2")))
+      diffUnreachable(g1, g2, dc2MemberDown.uniqueAddress) should ===(Seq())
+    }
+
+    "not be produced for unreachability observations from other dc" in {
+      val dc2MemberUp = TestMember(Address("akka.tcp", "sys", "otm", 2552), Up, Set.empty, "dc2")
+      val dc2MemberDown = TestMember(Address("akka.tcp", "sys", "otm", 2552), Down, Set.empty, "dc2")
+
+      val reachability1 = Reachability.empty
+      val g1 = Gossip(members = SortedSet(aUp, dc2MemberUp), overview = GossipOverview(reachability = reachability1))
+
+      val reachability2 = reachability1
+        .unreachable(dc2MemberUp.uniqueAddress, aRemoved.uniqueAddress)
+      val g2 = Gossip(members = SortedSet(aRemoved, dc2MemberUp), overview = GossipOverview(reachability = reachability2))
+
+      diffUnreachableTeam(g1, g2, aUp.uniqueAddress) should ===(Seq())
+    }
+
+    "not be produced for unreachability observations within other dc" in {
+      val dc2AMemberUp = TestMember(Address("akka.tcp", "sys", "dc2A", 2552), Up, Set.empty, "dc2")
+      val dc2AMemberDown = TestMember(Address("akka.tcp", "sys", "dc2A", 2552), Down, Set.empty, "dc2")
+
+      val dc2BMemberUp = TestMember(Address("akka.tcp", "sys", "dc2B", 2552), Up, Set.empty, "dc2")
+      val dc2BMemberDown = TestMember(Address("akka.tcp", "sys", "dc2B", 2552), Down, Set.empty, "dc2")
+
+      val reachability1 = Reachability.empty
+      val g1 = Gossip(members = SortedSet(aUp, dc2AMemberUp, dc2BMemberUp), overview = GossipOverview(reachability = reachability1))
+
+      val reachability2 = reachability1
+        .unreachable(dc2AMemberUp.uniqueAddress, dc2BMemberDown.uniqueAddress)
+      val g2 = Gossip(members = SortedSet(aUp, dc2AMemberUp, dc2BMemberDown), overview = GossipOverview(reachability = reachability2))
+
+      diffUnreachableTeam(g1, g2, aUp.uniqueAddress) should ===(Seq())
+    }
```

Comment:
Scenarios for 3 DCs with 2 nodes in each, {dc1-n1, dc1-n2}, {dc2-n1, dc2-n2}, {dc3-n1, dc3-n2}

a -> b is an observation of unreachability
selfDc = dc1

* dc1-n1 -> dc2-n1 => UnreachableDataCenter(dc2)
* dc1-n1 -> dc1-n2 => no events
* dc2-n1 -> dc2-n2 => no events
* dc3-n1 -> dc2-n1 => no events
* dc3-n1 -> dc3-n2 => no events
* dc2-n1 -> dc1-n1 => no events
* dc1-n1 -> dc2-n1, dc2-n2 -> dc2-n1 => UnreachableDataCenter(dc2)